src/transaction/TransactionController.ts

```diff
@@ -701,7 +714,13 @@ export class TransactionController extends BaseController<
    *
    * @param transactionID - ID of the transaction to cancel
    */
-  async stopTransaction(transactionID: string) {
+  async stopTransaction(
+    transactionID: string,
+    gasValues?: GasPriceValue | FeeMarketEIP1559Values,
+  ) {
+    if (gasValues) {
+      validateGasValues(gasValues);
```

Comment:
The validation added for the custom gas price for `speedUpTransaction` is still missing in this function